### PR TITLE
ONS-UK-trade-in-goods-by-industry-country-and-commodity 

### PR DESCRIPTION
**Pull request by osamede20**
exports.py and imports.py scripts reworked to produce accurate csv output data.
TransformTrace annotation added to exports.py and imports.py scripts.
main.py reworked to correctly call up exports.py and imports.py, and produce exports and imports csv output data respectively.

As transform requirements are not 'clear', an optional scraper, concat_main.py, was written to concatenate exports and imports datacubes, and produce a single csv output data. 
This scraper does not require exports.py and imports.py.

https://github.com/GSS-Cogs/family-trade/issues/3

### CODE REVIEW

File: datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    trace.start(title1, tab, columns, distribution1.downloadURL)
+    trace.start(title2, tab, columns, distribution2.downloadURL)
+    
+    if tab.name == 'tig_ind_ex':
+        flow = 'exports'
+    elif tab.name == 'tig_ind_im':
+        flow = 'imports'
+    print(flow)   
+ 
+    country = tab.filter('country').fill(DOWN).is_not_blank()
```
Comment (CharlesRendle) [inline datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py:74]: Good idea to add tracing comments for each dimension so that we can see where each value has been found in the dataset. E.g:
`trace.country("Values taken from the cell found by filtering entire tab for 'country' (A1) downwards")`

File: datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    info = json.load(file)
+
+landingPage = info['landingPage']
+landingPage
+
+# +
+scraper1 = Scraper(landingPage[0])
+scraper2 = Scraper(landingPage[1])
+
+scraper1.dataset.family = info['families']
```
Comment (CharlesRendle) [inline datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py:36]: We don't need to define the dataset family in the transform.

File: datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import json
+from gssutils import *
+
+pd.options.mode.chained_assignment = None 
+
+cubes = Cubes("info.json")
+
+# +
+with open ('info.json') as file:
+    info = json.load(file)
```
Comment (CharlesRendle) [inline datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py:27]: Can shorten this to: `info = json.load(open('info.json'))`
Not a huge necessity but can be neat to have it on a single line.

File: datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
+# ---
```
Comment (CharlesRendle) [inline datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py:1]: Based on my recommendation, if you choose to make this the main script, remember to change the name of this file to `main.py`

**Review by CharlesRendle — COMMENTED** [review]
This is cool! I see what you mean about doing it both ways. I think that you should go with `concat_main.py` as this outputs just the single csv file which is what we will want in the end. I have just commented on `concat_main.py`but if you wanted to have the two `imports.py` and `exports.py` then you will need to append the two returned dataframes at the end of the loop in 'main.py'. Remember though that `cubes.output_all()` should be the last line of the transform so shouldn't be inside the loop. Or go with `concat_main.py` in which case remember to rename this one to `main.py`

The branch has commits from HMRC Alcohol Bulletin. These need to be removed from the branch before merging.

Overall, happy with the output csv (of `concat_main.py`). Good job!

File: datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py (hunk 4 above)
Comment (osamede20) [reply datasets/ONS-UK-trade-in-goods-by-industry-country-and-commodity/concat_main.py:1]: Thanks Charles for reviewing the Pull Request. Yes, it is best to go with concat_main.py and change the file name to main.py.
